### PR TITLE
Add examples to functions in datasets.utils

### PR DESCRIPTION
**Pull request by Astro-Kirsty**
This PR is to add examples to the docstrings of the functions in `gammapy.datasets/utils`.

It also exposes the functions that previously were not -- do we want to expose these straight away?

This is connected to https://github.com/gammapy/gammapy/issues/5016

(I ran the docs and they look good!)

### CODE REVIEW

File: gammapy/datasets/utils.py [hunk 1]
```diff
@@ -14,13 +16,43 @@ def apply_edisp(input_map, edisp):
...
+    >>>
+    >>> axis = MapAxis.from_energy_bounds("1 TeV", "10 TeV", nbin=6, name="energy_true")
+    >>> m = Map.create(
+    ...     skydir=(0.8, 0.8),
+    ...     width=(1, 1),
+    ...     binsz=0.02,
+    ...     axes=[axis],
+    ...     frame="galactic"
+    ... )
+    >>> m.data = np.arange(m.data.size, dtype=float).reshape(m.geom.data_shape)
```
Comment (bkhelifi) [inline gammapy/datasets/utils.py:42]: Why this line? This is super counter intuitive... and geeky!

Comment (Astro-Kirsty) [reply gammapy/datasets/utils.py]: This is not actually needed, I accidentally left it there when I was testing other things. I have removed it!

**Review by bkhelifi — APPROVED** [review]
Thanks a lot @Astro-Kirsty ... Looks good for me!

**Comment by Astro-Kirsty** [comment]
I will rebase this as we have adapted the docs PR #5193

**Comment by Astro-Kirsty** [comment]
Ready for review!

**Pull request merged**